src/basic/fs-util.c

```diff
@@ -721,6 +722,9 @@ int chase_symlinks(const char *path, const char *original_root, unsigned flags,
 
                 if (fstat(child, &st) < 0)
                         return -errno;
+                if ((flags & CHASE_NO_AUTOFS) &&
+                    fd_check_fstype(child, AUTOFS_SUPER_MAGIC) > 0)
+                        return -EWOULDBLOCK;
```

Comment:
EWOULDBLOCK is the same as EAGAIN, and we usually call it EAGAIN, and hence we should do that here too... 

That said, I think we should pick a more exotic error code for this case, after all we propagate kernel errors, and EAGAIN feels like something the kernel might pass to us under some cases and we might not want to be confused by that. Maybe EREMOTE or so?